mono/metadata/boehm-gc.c

```diff
@@ -946,11 +948,21 @@ mono_gc_free_fixed (void* addr)
 	GC_FREE (addr);
 }
 
+static int validate_heap = 0;
+static int counter = 0;
+static int validate_frequency = 100;
+
+extern void mono_unity_heap_validation ();
 void *
 mono_gc_alloc_obj (MonoVTable *vtable, size_t size)
 {
 	MonoObject *obj;
 
+	if (validate_heap && (counter++ % validate_frequency) == 0)
```

Comment:
if you wanted to trim a variable could probably use the frequency to turn heap validation on/off.